doc/DataPlaneCodeGeneration/Handle-Ambiguity.md

```diff
@@ -0,0 +1,225 @@
+{
+    ...
+}
+
+public virtual async Task<Response> OperationAsync(string requiredId, string optionalId, RequestContext context) // Protocol method
+{
+    ...
+}
+```
+2. Append `Value` to the name of the **convenience method**.
+```C#
+public virtual async Task<Response<Model>> OperationValueAsync(string requiredId, string optionalId = null, CancellationToken cancellationToken = default) // Convenience method
+{
+    ...
+}
+
+public virtual async Task<Response> OperationAsync(string requiredId, string optionalId = null, RequestContext context = null) // Protocol method
+{
+    ...
+}
+```
+
+Our strategy is:
+- If the protocol method hasn’t been GAed, the first solution is taken if there exists an ambiguous call.
+- If the protocol method is already GAed, the second solution is taken because we cannot change protocol method anymore.
+
+## Scope of ambiguity we handle
+
+Not all the pairs of protocol method and convenience method will have such problem. Below is a case when there is no ambiguity if users are calling normally by passing a `model` or a `content`. (They could not pass in `null` because it will throw error)
+```C#
+public virtual async Task<Response<Model>> OperationAsync(Model model, CancellationToken cancellationToken = default) // Convenience method
+{
+    Argument.AssertNotNull(model, nameof(model));
+    ...
+}
+
+public virtual async Task<Response> OperationAsync(RequestContent content, RequestContext context = null) // Protocol method
+{
+    Argument.AssertNotNull(content, nameof(content));
+    ...
+}
+```
+Not all the ambiguities could be resolved by the first solution. For example, even we handle the example in **Background** section as follows
+```C#
+public virtual async Task<Response<Model>> OperationAsync(string requiredId, string optionalId = null, CancellationToken cancellationToken = default) // Convenience method
+{
+    ...
+}
+
+public virtual async Task<Response> OperationAsync(string requiredId, string optionalId, RequestContext context) // Protocol method
+{
+    ...
+}
+```
+`OperationAsync("required", "optional", default)` and `OperationAsync("required", "optional", new())` are two examples that still lead to compilation error of ambiguity.
+
+Considering all of these, we only guarantee that we will not generate code which leads to ambiguous call when:
+1. The call doesn't have any actual parameter of `null`, `default`, `new()`. For example, we don't care whether `OperationAsync("required", "optional", default)` will lead to ambiguity. Details [here](#special-values-not-taken-into-ambiguity-consideration).
+2. The call only has its required parameters passed in. For example, we can only ensure `OperationAsync("required")` is not an ambiguous call.
+
+See below section for more specific examples and corresponding explanation.
+
+## Examples
+
+### Operation without input
```

Comment:
the word input is quite ambiguous. `requiredId` is also an input in the following example.
Should it be `Operations without request body`?